libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c

```diff
@@ -1480,7 +1480,14 @@ static void prvOTAUpdateTask( void * pvUnused )
                                             ( void ) prvSetImageStateWithReason( eOTA_ImageState_Rejected, kOTA_Err_ImageStateMismatch );
                                             ( void ) prvResetDevice(); /* Ignore return code since there's nothing we can do if we can't force reset. */
                                         }
-                                    }
+									}
+                                    else if (OTA_GetImageState() == eOTA_ImageState_Unknown)
```

Comment:
Instead of calling prvSetImageStateWithReason after checking for eOTA_ImageState_Unknown we can call it for any other state than eOTA_ImageState_Testing. For example previous abort set the image state to eOTA_ImageState_Aborted and after that we get a job document that fails parsing.

Reply:
Updated the pull request to not check for this specific reason, but rather anything else other than "eOTA_ImageState_Testing" will be handled.